### PR TITLE
Refactor match change code

### PR DESCRIPTION
**Pull request by qinluhe**
### Feature Preview

- [x] refactor code that receive document change data

---

#### PR Checklist

- [ ] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  },
+  isRemote?: boolean
+) {
+  for (let i = 0; i < matchCases.length; i += 1) {
+    const matchCase = matchCases[i];
+    if (matchCase.match(command, path)) {
+      matchCase.onMatch(state, id, value, isRemote);
+      return;
+    }
+  }
```
Comment (LucasXu0) [inline frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts:37]:
```suggestion
  const matchCase = matchCases.find((e => e.match(command, path)));
  if (matchCase) {
    matchCase.onMatch(state, id, value, isRemote);
  }
```

Comment (LucasXu0) [reply frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts:37]: It looks more simple using `find`

File: frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      data: node.data,
+    };
+  }
+  return;
+}
+
+function onMatchBlockDelete(state: DocumentState, blockId: string, blockValue: BlockPBValue, _isRemote?: boolean) {
+  if (state.selections.indexOf(blockId)) {
+    state.selections.splice(state.selections.indexOf(blockId), 1);
+  }
```
Comment (LucasXu0) [inline frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts:133]:
```suggestion
  const index = state.selections.indexOf(blockId);
  if (index !== -1) {
    state.selections.splice(index, 1);
  }
```

File: frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts (hunk 1 above)
Comment (appflowy) [reply frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts]: If there is only one can response to that command then, I think find is better

File: frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts (hunk 2 above)
Comment (appflowy) [reply frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts]: LGTM

File: frontend/appflowy_tauri/src/appflowy_app/utils/tool.ts [hunk 3]
```diff
@@ -1,30 +1,30 @@
 export function debounce(fn: (...args: any[]) => void, delay: number) {
   let timeout: NodeJS.Timeout;
   return (...args: any[]) => {
-    clearTimeout(timeout)
-    timeout = setTimeout(()=>{
+    clearTimeout(timeout);
+    timeout = setTimeout(() => {
       // eslint-disable-next-line prefer-spread
```
Comment (appflowy) [inline frontend/appflowy_tauri/src/appflowy_app/utils/tool.ts:6]: Can we follow this guide: https://eslint.org/docs/latest/rules/prefer-spread

**Pull request merged**